### PR TITLE
v6 — Improve perfs

### PR DESCRIPTION
**Pull request by ooflorent**
This PR improves the performance of various utilities. These perf optimizations were previously done by `lodash`.

- Fix broken `every` tests (8590d4a6c32996326da942a39de4a450ff05bc80)
- Modify `toPath` to accept `null` literals (b1bbbe527966b78ab78b4d73d84b33e61c92250a)
- Remove closures created a runtime in `every` and `mapValues` and `toPath`

### CODE REVIEW

File: src/util/every.js [hunk 1]
```diff
@@ -2,12 +2,21 @@
  * Returns true of the predicate is true for every (value, key) pair in the Object or array
  */
 export default function every(object, predicate) {
-  if(!object) {
+  if (object !== Object(object)) {
```
Comment (ooflorent) [inline src/util/every.js]: Only arrays and objects will pass the test.

File: src/util/mapValues.js [hunk 2]
```diff
@@ -2,10 +2,18 @@
...
  */
 const mapValues = (obj, fn) => {
-  return obj ? Object.keys(obj).reduce((accumulator, key) => ({
-    ...accumulator,
-    [key]: fn(obj[key], key)
-  }), {}) : obj
+  if (obj !== Object(obj)) {
+    return obj
+  } else {
+    const acc = {}
```
Comment (ooflorent) [inline src/util/mapValues.js]: Previous version was extensively creating `accumulator` objects.
It's ok to mutate `acc`...

File: src/util/toPath.js [hunk 3]
```diff
@@ -1,7 +1,15 @@
...
 /**
  * Converts a string in dot-and-bracket notation to an array of keys
  */
-const toPath = value =>
-  value === undefined ? [] : String(value).split(/[.\[\]]/).filter(token => token.length)
+const toPath = value => {
+  if (value === undefined || value === null) {
+    return []
+  } else {
+    return String(value).split(DELIMITER_PATTERN).filter(isNotEmpty)
```
Comment (ooflorent) [inline src/util/toPath.js]: Hoist static declarations

**Comment by erikras** [comment]
All good stuff. Thanks!

**Pull request merged**